File: test-db.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const dbPath = path.join(process.cwd(), 'data', 'adani-excel.db');
console.log('Opening DB at:', dbPath);

try {
    const db = new sqlite3.Database(dbPath, (err) => {
        if (err) {
            console.error('Error opening DB:', err);
        } else {
            console.log('DB opened successfully');
            db.get('SELECT 1', (err, row) => {
                if (err) {
                    console.error('Error running query:', err);
                } else {
                    console.log('Query result:', row);
                }
            });
        }
    });
} catch (e) {
    console.error('Exception:', e);
}

File: test_db.py
import sys
import os
sys.path.append(os.path.join(os.path.dirname(__file__), 'backend'))

from backend.database import get_db_connection
import json
import sqlite3

def test_db_connection():
    try:
        conn = get_db_connection()
        print("Database connection successful")
        
        conn.row_factory = sqlite3.Row
        cursor = conn.cursor()
        
        # Test the specific query
        fiscalYear = "FY_23"
        cursor.execute('SELECT * FROM table_data WHERE fiscal_year = ? AND is_deleted = 0', (fiscalYear,))
        row = cursor.fetchone()
        
        if row:
            print(f"Found data for {fiscalYear}")
            print(f"Row keys: {row.keys()}")
            data = json.loads(row['data'])
            print(f"Data parsed successfully, contains {len(data)} items")
        else:
            print(f"No data found for {fiscalYear}")
            
        conn.close()
        return True
    except Exception as e:
        print(f"Error: {e}")
        return False

if __name__ == "__main__":
    test_db_connection()